src/File.php

```diff
@@ -741,6 +736,14 @@ protected function onBeforeWrite()
         parent::onBeforeWrite();
     }
 
+    /**
+     * Generate a readable title, dashes and underscores replaced by whitespace, and any file extensions removed.
+     */
+    public static function normaliseFileName(string $name): string
```

Comment:
```suggestion
    public static function getNormalisedFileName(string $name): string
```
This isn't setting the file name, which the name implies. Use `get` prefix to indicate we're getting a value.